 Settings: Fix cannot search paired BT device issue.
Fix can't search paired device if press search key in BT and then input key word.
We should delete the database data before we add something into
it.

diff --git a/src/com/android/settings/bluetooth/BluetoothDevicePreference.java b/src/com/android/settings/bluetooth/BluetoothDevicePreference.java
index 0005526..00ec128 100644
--- a/src/com/android/settings/bluetooth/BluetoothDevicePreference.java
+++ b/src/com/android/settings/bluetooth/BluetoothDevicePreference.java
@@ -240,7 +240,7 @@ private void pair() {
             data.screenTitle = context.getResources().getString(R.string.bluetooth_settings);
             data.iconResId = R.drawable.ic_settings_bluetooth;
             data.enabled = true;
-
+            Index.getInstance(context).deleteIndexableData(data);
             Index.getInstance(context).updateFromSearchIndexableData(data);
         }
     }
